### ISSUE TITLE
Schema broken in legacy AAT

### ISSUE THREAD

**Issue by swashbuck** [adaptlearning/adapt-contrib-pageLevelProgress#191]
### Subject of the issue
Page Level Progress no longer works in the legacy AAT. The browser console error is:

```
TypeError: _globals._extensions._pageLevelProgress is undefined
```

![console](https://github.com/adaptlearning/adapt-contrib-pageLevelProgress/assets/898168/d81055b3-4464-4b45-a12c-b72b814b3d6d)

Note that using "Force rebuild" fixes it, but then it's breaks again when using the "Preview course" option.

### Your environment
* Legacy AAT v0.10.5 using framework v5.31.17
* PLP v7.1.1

### Related
It seems that the following extensions may also not be included when using "preview course":
- _glossary
- _pageLevelProgress
- _resources
- _trickle
- _tutor

**Comment by taylortom** [comment]
Ok, so I've had a proper look at the code for this, and I think the situation is as follows:

From the AAT side, the only time defaults are stored in the DB is via the UI (i.e. the form lib we use makes sure these are added to the data when saved). At no point does the back-end add any missing defaults. Hence to properly fix this, we'd need to manually go into project settings and re-save.

As Ollie mentioned, clicking preview skips the grunt process most of the time (with a few exceptions) - on 'force rebuild', the grunt process will populate the defaults hence why things are working in this scenario.

To summarise:
- **Manual fix**: re-save the project settings for each course
- **AAT fix**: would require either modifying the build process to run the grunt task for all types of build, or some kind of back-end fix to ensure that defaults are applied.

**Comment by swashbuck** [comment]
Thanks, @taylortom .

It doesn't appear that resaving project settings fixes this issue. I still get the error when I resave settings and then view the course via Preview Course. The error appears in the browser console after **clicking the PLP button in the navigation**.

I'm looking at _course.json_ via Firefox's dev tools. Network tab > select course.json from the list > view the Response JSON.

```
  "_globals": {
    "_accessibility": {
      "skipNavigationText": "Skip navigation",
      "_ariaLabels": {
        "answeredIncorrectly": "You answered incorrectly",
        "answeredCorrectly": "You answered correctly",
        "selectedAnswer": "selected",
        "unselectedAnswer": "not selected",
        "skipNavigation": "Skip Navigation",
        "previous": "Back",
        "navigationDrawer": "Open course resources.",
        "close": "Close",
        "closeDrawer": "Close Drawer",
        "closeResources": "Close resources",
        "drawer": "Top of side drawer",
        "closePopup": "Close popup",
        "next": "Next",
        "done": "Done",
        "complete": "Completed",
        "incomplete": "Incomplete",
        "incorrect": "Incorrect",
        "correct": "Correct",
        "locked": "Locked",
        "visited": "Visited"
      }
    },
    "_extensions": {
      "_drawer": {
        "_navOrder": 100,
        "_navTooltip": {
          "_isEnabled": true,
          "text": "{{ariaLabel}}"
        }
      },
      "_navigation": {
        "_skipButton": {
          "_navOrder": -100
        },
        "_backButton": {
          "_navOrder": 0
        },
        "_backNavTooltip": {
          "_isEnabled": true,
          "text": "{{ariaLabel}}"
        },
        "_spacers": [
          {
            "_navOrder": 0
          }
        ]
      }
    },
```

Note that _globals.extensions_  does not contain the _pageLevelProgress_ object.

**Comment by swashbuck** [comment]
@taylortom This appears to be working now after an AAT upgrade to `0.11.2`. I'll close this issue.
